src/CheckForSafeFunctionCall.ts

```diff
@@ -1,101 +1,136 @@
+            }
+
+            var currentRegion = (
+                $arrayAccess.arrayVar.getDescendantsAndSelf(
+                    "varref"
+                )[0] as Varref
+            ).vardecl.currentRegion;
+            if (
+                (currentRegion != undefined &&
+                    currentRegion.joinPointType == "file") ||
+                (
+                    $arrayAccess.arrayVar.getDescendantsAndSelf(
+                        "varref"
+                    )[0] as Varref
+                ).vardecl.isParam
+            ) {
+                new_safefunctionCallslist.splice(
+                    new_safefunctionCallslist.indexOf($function.name),
+                    1
+                );
+            }
         }
-        
-    /*
-        if (
-            $memberAccess.use.indexOf('write') !== -1 &&
-            $memberAccess.getDescendantsAndSelf('varref')[0].vardecl.currentRegion.joinPointType === 'file'
-            )
-            new_safefunctionCallslist.splice(new_safefunctionCallslist.indexOf($function.name),1);		
-    */
-    end
-    condition new_safefunctionCallslist.indexOf($function.name) !== -1 end
+    }
 
+    for (const $function of Query.search(FileJp)
+        .search(FunctionJp)
+        .search(Body)
+        .search(MemberAccess)) {
+        if (new_safefunctionCallslist.indexOf($function.name) != -1) {
+            if ($memberAccess.use.indexOf("write") == -1) {
+                continue;
+            }
 
-    select file.function.body.varref end
-    apply
-        if($varref.useExpr.use.indexOf('write') === -1){
-            continue;
-        }
-    
-        var currentRegion = $varref.vardecl.currentRegion;
-        if(currentRegion !== undefined && currentRegion.joinPointType === 'file') {
-            new_safefunctionCallslist.splice(new_safefunctionCallslist.indexOf($function.name),1);
+            var currentRegion = (
+                $memberAccess.getDescendantsAndSelf("varref")[0] as Varref
+            ).vardecl.currentRegion;
+            if (
+                currentRegion !== undefined &&
+                currentRegion.joinPointType === "file"
+            ) {
+                new_safefunctionCallslist.splice(
+                    new_safefunctionCallslist.indexOf($function.name),
+                    1
+                );
+            }
         }
-        /*
-        if (
-            $varref.useExpr.use.indexOf('write') !== -1 &&
-            $varref.vardecl.currentRegion.joinPointType === 'file'
-            )
-            new_safefunctionCallslist.splice(new_safefunctionCallslist.indexOf($function.name),1);		
-        */
-    end
-    condition new_safefunctionCallslist.indexOf($function.name) !== -1 end
+    }
 
-    select file end
-    apply
-        $file.insertBegin('//new_safefunctionCallslist : ' + new_safefunctionCallslist.join(' , '));
-    end
+    for (const $function of Query.search(FileJp)
+        .search(FunctionJp)
+        .search(Body)
+        .search(Varref)) {
```

Comment:
```suggestion
    for (const $function of Query.search(FileJp).search(FunctionJp)) {
        for (const $varref of Query.searchFrom($function.body, Varref)) {
```

Depois precisas de um `}` extra no fim